src/backend/distributed/commands/parameter.c

```diff
@@ -0,0 +1,210 @@
+											  granteeOid,
+											  char *parameterName, AclMode mode,
+											  char *modeStr);
+static void RemoveSemicolonFromEnd(char *query);
+
+
+List *
+PostprocessGrantParameterStmt(Node *node, const char *queryString)
+{
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	EnsurePropagationToCoordinator();
+
+	char *command = DeparseTreeNode(node);
+
+	List *commands = list_make3(DISABLE_DDL_PROPAGATION,
+								(void *) command,
+								ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(REMOTE_NODES, commands);
+}
+
+
+/*
+ * GenerateGrantOnParameterFromAclItem generates the grant queries for the given aclItem.
+ * First it sets the current role to the grantor of the aclItem, then it appends the grant
+ * privilege queries for the aclItem, and finally it resets the role to the original role.
+ * Ex: If the aclItem has the grant option for ACL_SET, it generates the following queries:
+ * SET ROLE <grantor>;
+ * GRANT SET ON <parameterName> TO <grantee>;
+ * RESET ROLE;
+ */
+static List *
+GenerateGrantOnParameterFromAclItem(char *parameterName, AclItem *aclItem)
+{
+	/*
+	 * seems unlikely but we check if there is a grant option in the list without the actual permission
+	 */
+	ValidatePermissionsAndGrants(aclItem, (AclMode[]) { ACL_SET, ACL_ALTER_SYSTEM }, 2);
+	Oid granteeOid = aclItem->ai_grantee;
+	List *queries = NIL;
+
+	queries = lappend(queries, GenerateSetRoleQuery(aclItem->ai_grantor));
+
+	CheckAndAppendGrantParameterQuery(&queries, aclItem, granteeOid, parameterName,
+									  ACL_SET, "SET");
+	CheckAndAppendGrantParameterQuery(&queries, aclItem, granteeOid, parameterName,
+									  ACL_ALTER_SYSTEM,
+									  "ALTER SYSTEM");
+
+	queries = lappend(queries, "RESET ROLE");
+
+	return queries;
+}
+
+
+/*
+ * CheckAndAppendGrantParameterQuery checks if the aclItem has the given mode and if it has, it appends the
+ * corresponding query to the queries list.
+ * Ex: If the mode is ACL_SET, it appends the query "GRANT SET ON <parameterName> TO <grantee>"
+ */
+static void
+CheckAndAppendGrantParameterQuery(List **queries, AclItem *aclItem, Oid granteeOid,
+								  char *parameterName,
+								  AclMode mode, char *modeStr)
+{
+	AclResult aclresult = pg_parameter_aclcheck(parameterName, granteeOid, mode);
+	if (aclresult == ACLCHECK_OK)
+	{
+		char *query = DeparseTreeNode((Node *) GenerateGrantStmtForRightsWithObjectName(
+										  OBJECT_PARAMETER_ACL, granteeOid, parameterName,
+										  modeStr,
+										  HasAclGrantOption(aclItem, mode)));
+
+		RemoveSemicolonFromEnd(query);
+
+		*queries = lappend(*queries, query);
+	}
+}
+
+
+/*
+ * RemoveSemicolonFromEnd removes the semicolon at the end of the query if it exists.
+ */
+static void
+RemoveSemicolonFromEnd(char *query)
+{
+	/* remove the semicolon at the end of the query since it is already */
+	/* appended in metadata_sync phase */
+	if (query[strlen(query) - 1] == ';')
+	{
+		query[strlen(query) - 1] = '\0';
+	}
+}
```

Comment:
it's okay to have multiple `;` at the end of the query, so maybe just remove this because we're anyway experiencing this in other code-paths too